### PR TITLE
refactor: scaffold css framework (part1)

### PR DESCRIPTION
**Pull request by carlosescri**
Related to #37

### CODE REVIEW

File: packages/info/src/styles.ts [hunk 1]
```diff
@@ -1,9 +1,11 @@
 import { css } from 'lit-element';
+import { DS } from '@dfwc/core';
 
 export const styles = [
   css`
     :host {
       display: inline-block;
+      font-size: ${DS.$fontSize};
```
Comment (carlosescri) [inline packages/info/src/styles.ts:8]: This produces:

image

File: packages/info/src/styles.ts [hunk 2]
```diff
@@ -1,9 +1,14 @@
...
+import { DS, cssTokenFactory } from '@dfwc/core';
+
+const t = cssTokenFactory('--df-info');
 
 export const styles = [
   css`
     :host {
       display: inline-block;
+      font-size: ${DS.$fontSize};
+      color: ${t('color', DS.$primary)};
```
Comment (carlosescri) [inline packages/info/src/styles.ts:11]: This produces:

image

- You can change the color specifically for the component by setting `--df-info--color`.
- You can change the primary color for ALL components by setting `--df-theme-primary`.
- If you want to completely change the colors used for different variants of gray, just change the desired vars:

```
:root {
  --df-theme--gray000: #fff;
  …
}
```

File: playground/misc/inventory.html [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      --df-theme--primary-000: #f5f0ff;
+      --df-theme--primary-100: #e6dcfd;
+      --df-theme--primary-200: #d1bcf9;
+      --df-theme--primary-300: #b392f0;
+      --df-theme--primary-400: #8a63d2;
+      --df-theme--primary-500: #6f42c1;
+      --df-theme--primary-600: #5a32a3;
+      --df-theme--primary-700: #4c2889;
+      --df-theme--primary-800: #3a1d6e;
+      --df-theme--primary-900: #29134e;
```
Comment (carlosescri) [inline playground/misc/inventory.html:25]: Before:

image

After:

image

To get the desired effect we could just have done:

```css
df-info {
  --df-info--color: #6f42c1;
}
```

Or, if we just want to change the primary:

```css
:root {
  --df-theme--primary: #6f42c1;
}
```

or:

```css
:root {
  --df-theme--primary-500: #6f42c1;
}
```

But the rest of colors would remain blue. It depends on what you want to achieve.

File: playground/misc/inventory.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+<body data-df-hashid="7a6100c782c09126479e9270ecc619b3" data-df-server="eu1-search.doofinder.com">
+  <p>
+    <a href="../index.html">← Playground</a>
+  </p>
+
+  <h1>Inventory</h1>
+
+  <df-input-capture class="m-b-1">
+    <input type="search" placeholder="Search">
+    <df-info key="total_found"></df-info>
```
Comment (carlosescri) [inline playground/misc/inventory.html:40]: Default CSS generated for `<df-info>`:

```css

:host {
  display: inline-block;
  font-size: var(--df-theme--font-size, 14px);
  color: var(--df-info--color, var(--df-theme--primary, var(--df-theme--primary-500, #0366d6)));
  padding: var(--df-info--padding, var(--df-theme--spacer, var(--df-theme--spacer-2, 8px)));
}
```

File: packages/core/src/widget.ts [hunk 5]
```diff
@@ -22,10 +23,6 @@ const BASE_STYLES: CSSResult = css`
     visibility: hidden !important;
```
Comment (sonic182) [inline packages/core/src/widget.ts:23]: it's not for this pr but... can we avoid these important

Comment (carlosescri) [reply packages/core/src/widget.ts:23]: why?

**Pull request merged**